qutip/solver/brmesolve.py

```diff
@@ -0,0 +1,212 @@
+        either an array of expectation values, for operators given in e_ops,
+        or a list of states for the times specified by `tlist`.
+
+    .. note:
+        operator_data_type
+    """
+    H = QobjEvo(H, args=args, tlist=tlist)
+
+    c_ops = c_ops if c_ops is not None else []
+    if not isinstance(c_ops, (list, tuple)):
+        c_ops = [c_ops]
+    c_ops = [QobjEvo(c_op, args=args, tlist=tlist) for c_op in c_ops]
+
+    new_a_ops = []
+    for (a_op, spectra) in a_ops:
+        if isinstance(spectra, str):
+            new_a_ops.append(
+                (a_op, coefficient(spectra, args={**args, 'w':0})))
+        elif isinstance(spectra, InterCoefficient):
+            new_a_ops.append((a_op, SpectraCoefficient(spectra)))
+        elif isinstance(spectra, Coefficient):
+            new_a_ops.append((a_op, spectra))
+        elif callable(spectra):
+            sig = inspect.signature(spectra)
+            if tuple(sig.parameters.keys()) == ("w",):
+                spec = SpectraCoefficient(coefficient(spectra))
+            else:
+                spec = coefficient(spectra, args={**args, 'w':0})
+            new_a_ops.append((a_op, spec))
+        else:
+            raise TypeError("a_ops's spectra not known")
+
+    solver = BRSolver(
+        H, new_a_ops, c_ops, options=options,
+        use_secular=(sec_cutoff>=0.), sec_cutoff=sec_cutoff,
+    )
+
+    return solver.run(psi0, tlist, e_ops=e_ops)
+
+class BRSolver(Solver):
+    """
+    Bloch Redfield equation evolution of a density matrix for a given
+    Hamiltonian and set of bath coupling operators.
+
+    Parameters
+    ----------
+    H : :class:`Qobj`, :class:`QobjEvo`
+        Possibly time-dependent system Liouvillian or Hamiltonian as a Qobj or QobjEvo.
+        list of [:class:`Qobj`, :class:`Coefficient`] or callable that can be
+        made into :class:`QobjEvo` are also accepted.
+
+    a_ops : list of (a_op, spectra)
+        Nested list of system operators that couple to the environment,
+        and the corresponding bath spectra.
+
+        a_op : :class:`qutip.Qobj`, :class:`qutip.QobjEvo`
+            The operator coupling to the environment. Must be hermitian.
+
+        spectra : :class:`Coefficient`
+            The corresponding bath spectra. As a `Coefficient` using an 'w'
+            args. Can depend on ``t`` only if a_op is a :class:`qutip.QobjEvo`.
+            :class:`SpectraCoefficient` can be used to conver a coefficient
+            depending on ``t`` to one depending on ``w``.
+
+        Example:
+            a_ops = [
+                (a+a.dag(), coefficient('w>0', args={'w':0})),
+                (QobjEvo([b+b.dag(), f(t)]), coefficient(g(t, w))),
+                (c+c.dag(), SpectraCoefficient(coefficient(array, tlist=ws))),
+            ]
+
+    c_ops : list of :class:`Qobj`, :class:`QobjEvo`
+        Single collapse operator, or list of collapse operators, or a list
+        of Lindblad dissipator. None is equivalent to an empty list.
+
```

Comment:
`use_secular` and `sec_cutoff` should be documented. Perhaps only `sec_cutoff` should be used to match `brsolve` above.